GenumKit/Stencil/Contexts.swift

```diff
@@ -80,13 +83,17 @@ extension StoryboardParser {
       if let scenes = storyboardsScenes[storyboardName] {
         sbMap["scenes"] = scenes
           .sort({$0.storyboardID < $1.storyboardID})
-          .map { (scene: Scene) -> [String:String] in
-            // Handle special scene.tag cases like navigationController, splitViewController, etc…
-            // TODO: Fix this to extract the 'UI' prefix and be able to implement OSX support in #128
-            let customClass = scene.customClass ??
-              (scene.tag != "viewController" ? "UI" + uppercaseFirst(scene.tag) : nil)
-            if let customClass = customClass {
-              return ["identifier": scene.storyboardID, "class": customClass]
+          .map { (scene: Scene) -> [String:AnyObject] in
+            if scene.customClass == nil && scene.tag == "viewController" {
+              return [
+                "identifier": scene.storyboardID,
+                "baseType": uppercaseFirst(scene.tag),
+                "isBaseViewController": true
+              ]
+            } else if scene.customClass == nil {
+                return ["identifier": scene.storyboardID, "baseType": uppercaseFirst(scene.tag)]
+            } else if let customClass = scene.customClass {
+              return ["identifier": scene.storyboardID, "customClass": customClass]
             } else {
               return ["identifier": scene.storyboardID]
```

Comment:
I don't see when that `else` case can happen. the two above conditions covers both `if scene.customClass == nil` and `if let customClass = scene.customClass` (which pass only if `scene.customClass != nil`) so by that time one of those two path should have been taken and the `else` doesn't cover any case left.

This also means that you can rewrite that whole `if` as a simple:

```swift
if let customClass = scene.customClass {
  return ["identifier": scene.storyboardID, "customClass": customClass]
} else {
  /* customClass is nil */
return [
    "identifier": scene.storyboardID,
    "baseType": uppercaseFirst(scene.tag)
    "isBaseViewController": (scene.tag == "viewController")
  ]
}
```

Which reduces from 4 conditions to only two and makes the whole conditional block more readable :wink: